dask/dataframe/core.py

```diff
@@ -309,6 +310,261 @@ def _scalar_binary(op, self, other, inv=False):
         return Scalar(graph, name, meta)
 
 
+class PartitionStatistics:
+    """
+    Container for DataFrame partition statistics
+
+    This class should be initialized with a dictionary
+    containing partition-wise statistics. Optional keys
+    include:
+      - "__num_rows__": Series of partition row counts
+      - <column-name>: DataFrame of stats for the column
+        - Required DataFrame columns: "min" and "max"
+
+    If any values in the input dictionary contain callable
+    objects, these statistics will be treated as lazy.
+    "Lazy" statistics correspond to partition statistics
+    that can be loaded "just in time" using a callback
+    function. The call-back function must accept an
+    optional `keys: set` argument to specify which columns
+    to load statistics for.
+    """
+
+    _statistics: dict
+
+    def __init__(
+        self,
+        statistics: dict | None = None,
```

Comment:
Yes - The `"__num_rows__"` magic key needs to go!

What I am trying to do here is come up with a way to both separate and couple the various types of partition statistics at the same time. This is because we want to be able to access both column and num-rows statistics individually, but we also want to be able to use the same callback (maybe delayed?) function to load multiple statistics "lazily" at the same time.

I am still struggling a bit to come up with an elegant solution for this.